llvm/lib/IR/DIBuilder.cpp

```diff
@@ -551,7 +555,7 @@ DIBuilder::createVariantPart(DIScope *Scope, StringRef Name, DIFile *File,
   auto *R = DICompositeType::get(
       VMContext, dwarf::DW_TAG_variant_part, Name, File, LineNumber,
       getNonCompileUnitScope(Scope), nullptr, SizeInBits, AlignInBits, 0, Flags,
-      Elements, 0, nullptr, nullptr, UniqueIdentifier, Discriminator);
+      Elements, 0, nullptr, nullptr, UniqueIdentifier, 0, Discriminator);
```

Comment:
This looks wrong. `DICompositeType::get` takes the new `NumExtraInhabitants` parameter at the end. Your change is now passing 0 for Discriminator, Discriminator for the DataLocation parameter, etc. That broke rustc's test suite, see (parts of) https://crbug.com/377743159

Since `DICompositeType::get`'s `NumExtraInhabitants` was added at the end, and with a default value, no change here is necessary.

I've followed up in https://github.com/llvm/llvm-project/commit/546066e4f74d50c974248b0ed247f65ebf24b75c

Reply:
Thanks @zmodem

Reply:
This happened [previously too](https://github.com/llvm/llvm-project/pull/72234#issuecomment-1813093440) with a similar change to the `::get` APIs. Wonder if it's worth doing something about this to catch these types of issues at compile time. E.g., using stronger types is my first instinct.